ddtrace/monkey.py

```diff
@@ -1,104 +1,19 @@
-"""Patch librairies to be automatically instrumented.
-
-It can monkey patch supported standard libraries and third party modules.
-A patched module will automatically report spans with its default configuration.
-
-A library instrumentation can be configured (for instance, to report as another service)
-using Pin. For that, check its documentation.
-"""
-import importlib
-import logging
-import sys
-import threading
-
-from wrapt.importer import when_imported
-
-
-log = logging.getLogger(__name__)
-
-# Default set of modules to automatically patch or not
-PATCH_MODULES = {
-    'asyncio': False,
-    'boto': True,
-    'botocore': True,
-    'bottle': True,
-    'cassandra': True,
-    'celery': True,
-    'elasticsearch': True,
-    'futures': False,  # experimental propagation
-    'grpc': True,
-    'mongoengine': True,
-    'mysql': True,
-    'mysqldb': True,
-    'pymysql': True,
-    'psycopg': True,
-    'pylibmc': True,
-    'pymemcache': True,
-    'pymongo': True,
-    'redis': True,
-    'requests': True,
-    'sqlalchemy': True,
-    'sqlite3': True,
-    'aiohttp': True,  # requires asyncio (Python 3.4+)
-    'aiopg': True,
-    'aiobotocore': True,
-    'httplib': True,
-    'vertica': True,
-    'jinja2': True,
-
-    # Even though instrumentation for web frameworks can be installed manually,
-    # instrumenting should be done idempotently
-    'django': True,
-    'falcon': True,
-    'pyramid': True,
-    'pylons': True,
-    # Ignore some web framework integrations that might be configured explicitly in code
-    "flask": False,
-}
-
-_LOCK = threading.Lock()
-_PATCHED_MODULES = set()
-
-# Modules which are patched on first use
-# DEV: These modules are patched when the user first imports them, rather than
-#      explicitly importing and patching them on application startup `ddtrace.patch_all(module=True)`
-# DEV: This ensures we do not patch a module until it is needed
-# DEV: <contrib name> => <list of module names that trigger a patch>
-_PATCH_ON_IMPORT = {
-    'celery': ('celery', ),
-    'gevent': ('gevent', ),
-    'requests': ('requests', ),
-}
-
-
-class PatchException(Exception):
-    """Wraps regular `Exception` class when patching modules"""
-    pass
-
-
-def _on_import_factory(module, raise_errors=True):
-    """Factory to create an import hook for the provided module name"""
-    def on_import(hook):
-        # Import and patch module
-        path = 'ddtrace.contrib.%s' % module
-        imported_module = importlib.import_module(path)
-        imported_module.patch()
-
-    return on_import
+from .install import patch as _patch, patch_all as _patch_all
+from .utils.deprecation import deprecated
 
 
+@deprecated(message='Use ddtrace.install.patch_all instead', version='1.0.0')
```

Comment:
I'd like to eventually move away from the word "patch" because not only does it feel hacky, but it also doesn't truly convey what we are doing. I think `install` would be a nicer alternative.

Reply:
`patch` is inline with what other libraries (e.g. `gevent` call it).

Reply:
Ah okay. I was unaware. If this is the convention then I'm fine sticking to it.

Reply:
I think given the direction we are heading in `install` is a better word for what we are doing, but `patch` is fine, and `patch` keeps us backwards compatible with what we have today.